ports/stm32/machine_pixelbitstream.c

```diff
@@ -0,0 +1,131 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/mphal.h"
+#include "modmachine.h"
+
+/*
+Example usage:
+
+import machine
+
+class NeoPixel:
+    ORDER = (1, 0, 2, 3)
+    TIMING = (350, 800, 700, 600)
+
+    def __init__(self, pin, n, bpp=3):
+        self.pin = pin
+        self.n = n
+        self.bpp = bpp
+        self.buf = bytearray(n * bpp)
+        self.pin.init(pin.OUT)
+
+    def __setitem__(self, index, val):
+        offset = index * self.bpp
+        for i in range(self.bpp):
+            self.buf[offset + self.ORDER[i]] = val[i]
+
+    def __getitem__(self, index):
+        offset = index * self.bpp
+        return tuple(self.buf[offset + self.ORDER[i]]
+                     for i in range(self.bpp))
+
+    def fill(self, color):
+        for i in range(self.n):
+            self[i] = color
+
+    def write(self):
+        machine.pixelbitstream(self.pin, self.TIMING, self.buf)
+
+
+np = NeoPixel(machine.Pin('X1', machine.Pin.OUT), 8)
+
+*/
+
+#if 0
+// STM32F091 @ 48MHz
+#define NS_PRESCALE (6)
+#define NS_OVERHEAD (1)
+#define NS_OVERHEAD_LOW (2)
+#else
+// Tested on PYBv1.0
+#define NS_PRESCALE (6)
+#define NS_OVERHEAD (0)
+#define NS_OVERHEAD_LOW (0)
+#endif
+
+uint32_t mp_hal_delay_ns_calc(uint32_t ns, uint32_t extra_overhead) {
+    uint32_t nclk = SystemCoreClock / 1000000 * ns / 1000 / NS_PRESCALE;
```

Comment:
I would like some brackets on this line, and maybe for performance, you could get rid of a division. Might just compile out anyways.